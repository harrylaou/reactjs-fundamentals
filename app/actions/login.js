export const LOGIN_USER_OK = 'LOGIN_USER_OK'
export const LOGOUT_USER = 'LOGOUT_USER'

export const loginUser = () => ({
  type: LOGIN_USER_OK
})

export const logoutUser = () => ({
  type: LOGOUT_USER
})