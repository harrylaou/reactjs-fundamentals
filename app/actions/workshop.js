export const RECEIVE_WORKSHOP = 'RECEIVE_WORKSHOP'


export const receiveWorkshop = workshop => ({
  type: RECEIVE_WORKSHOP,
  workshop
})