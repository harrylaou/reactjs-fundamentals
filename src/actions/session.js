export const LOG_OUT = 'LOG_OUT'
export const LOGGED_IN = 'LOGGED_IN'

export const logout = () => ({
  type: LOG_OUT
})

export const loggedIn = token => ({
  type: LOGGED_IN,
  token
})
