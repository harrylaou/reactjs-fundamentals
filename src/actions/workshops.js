export const RECEIVE_WORKSHOPS = 'RECEIVE_WORKSHOPS'

export const receiveWorkshops = workshops => ({
  type: RECEIVE_WORKSHOPS,
  workshops
})
