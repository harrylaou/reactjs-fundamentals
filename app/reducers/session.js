import { LOG_OUT, LOGGED_IN } from "../actions/session"
import { getSessionToken } from '../utils/localStorage'

// There is a better way to initialize the state with the store initial_state and the
// Redux middleware. We will refactor this when we explain you the Redux middleware
// Do you see any problem by doing this here? coupling? testing?
const token = getSessionToken()

const session = (state = { token }, action) => {
  switch (action.type) {
    case LOGGED_IN:
      return { ...state, token: action.token }
      break;
    case LOG_OUT:
      return { ...state, token: null }
      break;
    default:
      return state
  }
}

export default session
