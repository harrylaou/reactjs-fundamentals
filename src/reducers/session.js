import { LOG_OUT, LOGGED_IN } from "../actions/session"
import { getSessionToken } from '../utils/localStorage'

// There is a better way to initialize the state with the store initial_state and the
// Redux middleware. We will refactor this when we explain you the Redux middleware
// Do you see any problem by doing this here? coupling? testing?
const token = getSessionToken()

const session = (state = { token }, action) => {
  switch (action.type) {
    case LOGGED_IN:
      return { ...state, token: action.token }
    case LOG_OUT:
      return { ...state, token: null }
    default:
      return state
  }
}

export default session
