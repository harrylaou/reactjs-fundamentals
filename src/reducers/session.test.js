import session from './session'
import {LOG_OUT, LOGGED_IN} from '../actions/session'

describe('#Reducer: Session', () => {
  it('Return default state',() => {
    expect(session(undefined, { type: 'default_action' })).toEqual({ token: undefined })
    expect(session(undefined, { type: 'default_action' })).toMatchSnapshot()
  })

  it('Return state',() => {
    expect(session({ token: true }, { type: 'another_action' })).toEqual({ token: true })
    expect(session({ token: true }, { type: 'another_action' })).toMatchSnapshot()
  })

  it(`When Action ${LOGGED_IN} return state`,() => {
    expect(session({ token: false }, { type: LOGGED_IN, token: true })).toEqual({ token: true })
    expect(session({ token: false }, { type: LOGGED_IN, token: true })).toMatchSnapshot()
  })

  it(`When Action ${LOG_OUT} return state`,() => {
    expect(session({ token: true }, { type: LOG_OUT })).toEqual({ token: null })
    expect(session({ token: true }, { type: LOG_OUT })).toMatchSnapshot()
  })
})