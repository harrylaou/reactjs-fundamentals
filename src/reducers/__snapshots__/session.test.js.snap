// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Reducer: Session Return default state 1`] = `
Object {
  "token": undefined,
}
`;

exports[`#Reducer: Session Return state 1`] = `
Object {
  "token": true,
}
`;

exports[`#Reducer: Session When Action LOG_OUT return state 1`] = `
Object {
  "token": null,
}
`;

exports[`#Reducer: Session When Action LOGGED_IN return state 1`] = `
Object {
  "token": true,
}
`;
